File: Swift/LeetCode/LeetCode3751.swift
// https://leetcode.com/problems/total-waviness-of-numbers-in-range-i

class Solution {
    func totalWaviness(_ num1: Int, _ num2: Int) -> Int {
        var answer = 0
        for num in num1...num2 {
            let numArray = String(num).compactMap { Int(String($0)) }
            guard numArray.count > 2 else { continue }
            for idx in 1..<numArray.count - 1 {
                if numArray[idx] > numArray[idx - 1] && numArray[idx] > numArray[idx + 1] {
                    answer += 1
                } else if numArray[idx] < numArray[idx - 1] && numArray[idx] < numArray[idx + 1] {
                    answer += 1
                }
            }
        }
        return answer
    }
}

File: C++/LeetCode/LeetCode3751.c
int totalWaviness(int num1, int num2) {
    int answer = 0;
    for (int num = num1; num <= num2; num++) {
        char buffer[20];
        int numArray[20];
        sprintf(buffer, "%d", num);
        int count = 0;
        for (int i = 0; buffer[i] != '\0'; i++) {
            numArray[i] = buffer[i] - '0';
            count++;
        }
        for (int i = 1; i < count - 1; i++) {
            if (numArray[i] > numArray[i - 1] && numArray[i] > numArray[i + 1]) {
                answer += 1;
            } else if (numArray[i] < numArray[i - 1] && numArray[i] < numArray[i + 1]) {
                answer += 1;
            }
        }
    }
    return answer;
}
